libqtile/core/manager.py

```diff
@@ -174,9 +174,38 @@ def load_config(self, initial: bool = False) -> None:
         if self.config.reconfigure_screens:
             hook.subscribe.screen_change(self.cmd_reconfigure_screens)
 
+        # If user wants wake_up hooks we need to add a dbus rule
+        if "wake_up" in hook.subscriptions:
+            task = asyncio.create_task(
+                add_signal_receiver(
+                    self.on_wake,
+                    session_bus=False,
+                    dbus_interface="org.freedesktop.login1.Manager",
+                    bus_name="org.freedesktop.login1",
+                    check_service=True,
+                )
+            )
+            task.add_done_callback(self._wake_callback)
+
         if initial:
             hook.fire("startup_complete")
```

Comment:
Is there any reason we must create the task before the startup_complete hook fires, or could you move it after? That way, if anybody is binding hooks within startup_complete, they will work.

Reply:
No reason at all.

Reply:
One reason might be if there is some state detectable from within the startup_complete hook that relates to whether or not that task has been sent. Not that I can think why anybody might be digging into that though!

Reply:
My logic was simpler: `startup_complete` should be called once everything has been set up. Therefore, any steps for adding hooks, which are part of starting up, should be done before that hook is fired.

Reply:
That makes a lot of sense, but puts a lot of faith in users :')